inst/www/shared/shiny.js

```diff
@@ -1260,106 +1279,801 @@
+          }
+
+          var offset = mouseOffset(e);
+          // Ignore events outside of plotting region
+          if (clip && !isInPlottingRegion(offset)) return;
+
+          var coords = offsetToScaledCoords(offset);
+          coords[".nonce"] = Math.random();
+          exports.onInputChange(inputId, coords);
+        };
+      }
+
+      function createClickHandler(inputId) {
+        var clickInfoSender =  mouseCoordinateSender(inputId, opts.clickClip);
+
+        return {
+          mousedown: function(e) {
+            // Listen for left mouse button only
+            if (e.which !== 1) return;
+            clickInfoSender(e);
+          },
+          onRemoveImg: function() { clickInfoSender(null); }
+        };
+      }
+
+      function createHoverHandler(inputId) {
+        var sendHoverInfo = mouseCoordinateSender(inputId, opts.hoverClip);
+
+        var hoverInfoSender;
+        if (opts.hoverDelayType === 'throttle')
+          hoverInfoSender = new Throttler(null, sendHoverInfo, opts.hoverDelay);
+        else
+          hoverInfoSender = new Debouncer(null, sendHoverInfo, opts.hoverDelay);
+
+        return {
+          mousemove:   function(e) { hoverInfoSender.normalCall(e); },
+          onRemoveImg: function()  { hoverInfoSender.immediateCall(null); }
+        };
+      }
+
+      // Returns a brush handler object. This has three public functions:
+      // mousedown, mousemove, and mouseup.
+      function createBrushHandler(inputId) {
+
+        // Object that encapsulates brush state
+        var brush = {
+          // Current brushing and dragging state
+          brushing: false,
+          dragging: false,
+
+          // Offset of last mouse down and up events
+          down: { x: NaN, y: NaN },
+          up: { x: NaN, y: NaN },
+
+          // Bounding rectangle of the brush
+          bounds: {
+            xmin: NaN,
+            xmax: NaN,
+            ymin: NaN,
+            ymax: NaN
+          },
+
+          // The bounds at the start of a drag
+          dragStartBounds: {
+            xmin: NaN,
+            xmax: NaN,
+            ymin: NaN,
+            ymax: NaN
+          },
+
+          // div that displays the brush
+          $div: null,
+
+          reset: function() {
+            this.brushing = false;
+            this.dragging = false;
+            this.down = { x: NaN, y: NaN };
+            this.up   = { x: NaN, y: NaN };
+            this.bounds = {
+              xmin: NaN,
+              xmax: NaN,
+              ymin: NaN,
+              ymax: NaN
+            };
+            this.dragStartBounds = {
+              xmin: NaN,
+              xmax: NaN,
+              ymin: NaN,
+              ymax: NaN
+            };
+
+            if (this.$div)
+              this.$div.remove();
+
+            return this;
+          },
+
+          // If there's an existing brush div, use that div to set the new
+          // brush's settings.
+          importOldBrush: function() {
```

Comment:
The brush still gets imported to the new image. This is because, although we can detect that a new image has been sent, we can't tell *why* it was sent, as mentioned previously.